connection.go

```diff
@@ -48,6 +49,26 @@ type mysqlConn struct {
 	finished chan<- struct{}
 	canceled atomicError // set non-nil if conn is canceled
 	closed   atomicBool  // set when conn is closed, before closech is closed
+
+	// for killing query after timeout
+	id int
+	d MySQLDriver
+	dsn string
+}
+
+func (mc *mysqlConn) killSelfQuery() error {
```

Comment:
what about name 'kill', like 'close', 'cleanup', 'cancel'

Reply:
Ok. Renamed to 'kill'.